### PR TITLE
Adding close button on Support menu

### LINKED ISSUE #5243: User get stuck in the Support menu
Some users will rely on tools such as a switch control (https://webaim.org/articles/motor/assistive) that only allow for moving forward or selecting. These users will be stuck within this menu and not be able to trigger esc to close it.
This bug is in order to keep the a11y guide line: [3.2 All functionality must be accessible to keyboard users without traps](https://a11y-demos-web.us-west-2.test.eginternaltools.expedia.com/pages/ExAG/3-2)

## Expected Behavior

The user should be able to close the support menu without the mouse or esc key. Which is not possible with the current implementation.

## Current Behavior

The only way to get out of the support menu without the mouse is the esc key, which not everyone has access to.

## Possible Solution

As a possible solution, we can add a close button to the support menu, so the user can close them with using the tab & enter key, like we see in the below image:
image

## Steps to Reproduce

Using the keyboard or a screen reader:
1. Go to https://backstage.expedia.biz/create
2. Navigate to the "Support" button and trigger
3. Attempt to exit the menu that opens

## Context

In Expedia Group, we care about keep our products align with the A11y guideline. So as we currently using Backstage, we think this can be an improvement that would help to all the community.

## Your Environment

- NodeJS Version (v12): v12.20.1
- Operating System and Version (e.g. Ubuntu 14.04): MacOs Catalina v10.15.7
- Browser Information: Chrome/89.0.4389.114

### PR DESCRIPTION
**Pull request by victormorfin97**
## Hey, I just made a Pull Request!

Adding a close button in Support menu, to be align with a11y guideline. Please check this [issue](https://github.com/backstage/backstage/issues/5243) to get more information .

Fixes #5243.

Screen Shot 2021-04-07 at 14 34 05

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/stale-chefs-retire.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core': minor
```
Comment (Rugvip) [inline .changeset/stale-chefs-retire.md:2]: This'll be a `patch` change as it's not breaking an we're still on 0.x

**Review by Rugvip — COMMENTED** [review]
Thanks, looks good! Just a minor changeset nit

Any input on the design @katz95?

File: .changeset/stale-chefs-retire.md (hunk 1 above)
Comment (victormorfin97) [reply .changeset/stale-chefs-retire.md:2]: Thanks, let me update the changeset

File: packages/core/src/components/SupportButton/SupportButton.tsx [hunk 2]
```diff
@@ -127,6 +128,11 @@ export const SupportButton = ({ children }: PropsWithChildren<Props>) => {
           {items &&
             items.map((item, i) => <SupportListItem item={item} key={i} />)}
         </List>
+        <Box display="flex" justifyContent="flex-end">
```
Comment (Rugvip) [inline packages/core/src/components/SupportButton/SupportButton.tsx:131]: While waiting for UX feedback, does using `DialogActions` instead work here? Feeling that it's a bit more specific and semantically correct

Comment (victormorfin97) [reply packages/core/src/components/SupportButton/SupportButton.tsx:131]: Yeah, works too. I'm agree with you that would be more specific, will commit the changes for that.

**Comment by victormorfin97** [comment]
@Rugvip is there any pending changes for this or what I missing in order to get this merged ? 😕

**Comment by freben** [comment]
I'm just getting some design input and then we're probably good to go!

**Comment by katz95** [comment]
the close button looks great!!! thanks for providing this suggestion!

**Review by freben — APPROVED** [review]

**Pull request merged**